XamarinApp.Test/ViewModels/LoginPageViewModelTest.cs

```diff
@@ -0,0 +1,232 @@
+
+      //Assert
+      _pageDialogService.Verify(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
+    }
+
+    [Fact]
+    public void GIven_Empty_Username_and_Password__Show_AlertBox()
+    {
+      //Arrange
+      var user = _fixture.Create<UserModel>();
+
+      //Act
+      _navigationService.Setup(x => x.NavigateAsync("MainPage")).ReturnsAsync(_fixture.Create<NavigationResult>());
+      _pageDialogService.Setup(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
+      _loginService.Setup(x => x.LoginUser(It.Is<UserModel>(x => x.UserName == "" && x.Password == ""))).ReturnsAsync(false);
+      viewModel.LoginCommand.Execute(new object());
+
+      //Assert
+      _pageDialogService.Verify(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
+    }
+
+    [Fact]
+    public void GIven_Valid_Userdetail_InValid_Password_Navigate_to_MainPage()
+    {
+      //Arrange
+      var user = _fixture.Create<UserModel>();
+
+      //Act
+      _navigationService.Setup(x => x.NavigateAsync("MainPage")).ReturnsAsync(_fixture.Create<NavigationResult>());
+      _pageDialogService.Setup(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
+      _loginService.Setup(x => x.LoginUser(It.Is<UserModel>(x => x.UserName == user.UserName && x.Password == "2000"))).ReturnsAsync(false);
+      viewModel.UserName = user.UserName;
+      viewModel.PassWord = "2000";
+
+      viewModel.LoginCommand.Execute(new object());
+
+      //Assert
+      _pageDialogService.Verify(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
+    }
+
+    [Fact]
+    public void GIven_InValid_Userdetail_Valid_Password_Navigate_to_MainPage()
+    {
+      //Arrange
+      var user = _fixture.Create<UserModel>();
+
+      //Act
+      //_navigationService.Setup(x => x.NavigateAsync("MainPage")).ReturnsAsync(_fixture.Create<NavigationResult>());
+      //_pageDialogService.Setup(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
+      _loginService.Setup(x => x.LoginUser(It.Is<UserModel>(x => x.UserName == "naveen" && x.Password == user.Password))).ReturnsAsync(false);
+      viewModel.UserName = "naveen";
+      viewModel.PassWord = user.Password;
+
+      viewModel.LoginCommand.Execute(new object());
+
+      //Assert
+      _pageDialogService.Verify(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
+    }
+
+    [Fact]
+    public void When_User_Click_SignUp_Navigate_to_SignUpPage()
+    {
+      //Arrange
+      var user = _fixture.Create<UserModel>();
+
+      //Act
+      viewModel.UserName = user.UserName;
+      viewModel.PassWord = user.Password;
+      _pageDialogService.Setup(x => x.DisplayAlertAsync("Request", "Do You want to Sign Up", "Yes", "No")).ReturnsAsync(true);
+      _pageDialogService.Setup(x => x.DisplayAlertAsync("Failed", "Incorrect Username or Password", "OK"));
```

Comment:
why your setups are missing return statements? check all the setups

Reply:
Sorry, That setups are not need in this test cases I have removed